### ISSUE TITLE
REQUEST: Apply to selected (Multi select objects to edit specific props)

### ISSUE THREAD

**Issue by Shlok-Bhakta** [KirilStrezikozin/BakeMaster-Blender-Addon#7]
**This feature request is:**
- [x] This feature request is not a duplicate
- [x] Implemented

**How about:**
- A way to **_apply a setting to all/multiple objects at once_** (through a checkbox system or a ctrl + click system, I am frustrated when I have 100+ objects that I want to apply settings to)

How I think it can be implemented:
- a checkbox system (not great but better than nothing)
- a way to multiselect with ctrl (selects specific objects) and shift (selects everything in the range) then click on settings and have them apply to all of those selected objects. (would also be nice to have a visual indicator that something has been selected)
- a button that you can press to sync all settings from the current object to everything
- use blenders copy all to selected option (would pair well with the second bullet point)
![image](https://user-images.githubusercontent.com/62256489/198910225-1f54cedb-681b-469e-993d-5d93113d3a37.png)

**Edited by @KirilStrezikozin**:
- [x] Add Operator to sync lastly edited property with chosen objects (Remove in [3.0.0](https://github.com/KirilStrezikozin/BakeMaster-Blender-Addon/tree/dev-3.0.0))

**Comment by KirilStrezikozin** [comment]
Hi, @Shlok-Bhakta

**1. Solution to apply settings to all objects or an ability to save settings to preset and quickly apply it to any object**:
    **Presets**: https://github.com/KirilStrezikozin/BakeMaster-Blender-Addon/issues/3 (solved)
    **How to get it**: https://github.com/KirilStrezikozin/BakeMaster-Blender-Addon/issues/3#issuecomment-1270204989
    To save and apply all object settings:
    ![image](https://user-images.githubusercontent.com/77115208/199049844-9a965cf9-a450-4771-8691-af9275832329.png)
    To apply to all objects, click on saved preset, at the bottom left corner of your screen, check the "Affect All Objects":
    ![image](https://user-images.githubusercontent.com/77115208/199050117-05b7843f-3c76-4c9e-9bef-9e740404fd07.png)

---

**2. About some ideas:**
    **first** point: what do you mean under `checkbox system`, please can you describe
    **second & fourth** points: I will think about the ability to apply specific setting from one object to other, that'd be cool, thank you
    **third** point: can be done through presets presented in **1.**

**Comment by Shlok-Bhakta** [comment]
Hello, this is a really quick mockup

https://user-images.githubusercontent.com/62256489/199217248-f3d0b1e5-0a26-4ea6-a93a-2e2401a86d60.mp4

I would disregard the checkboxes as I believe it is a terrible idea compared to the others, and I was not aware that you could apply all through presets.

For 2 and three combined the video sums up what would be nice to have (very quickly made though so its pretty bad)

**Comment by KirilStrezikozin** [comment]
Hi, great 👍 

Adding custom stuff to right-click menu is possible, but the problem here is that is hard-coded and I haven't found a way yet to add this custom stuff to only when pressed somewhere on BakeMaster setting.
#### But I'm thinking about the following **solution**:
- I can find out the setting that was lastly edited.
- I can create an operator where you can choose for which objects apply that setting
- I can put this operator somewhere in the UI, but I'd personally make it the way that user would run it through the search menu: 
![image](https://user-images.githubusercontent.com/77115208/199284060-098e2164-e53e-45b0-ad9a-a355ab05da81.png)

Tell me what you think about it)

**Comment by KirilStrezikozin** [comment]
Oh and another thing that would work with it then is:
if you need multiple settings synced with other objects, you won't need to choose those objects every time you run that "Sync operator", it will show lastly chosen objects. And an option to clear that and rechoose what objects to sync with.

**Comment by Shlok-Bhakta** [comment]
I am so sorry, I did not realize the selection method would not work. I love The solution that you have. It looks good!

**Comment by KirilStrezikozin** [comment]
hi @Shlok-Bhakta 👋
there's a lot going on in our [discord server](https://discord.gg/2ePzzzMBf4), and I wanted to announce that

> your request has been implemented

and will be released in the BakeMaster update, which is currently under hyperactive development 😁

Check out the discord server to try out all new features yourself, and **below** are a couple of **snapshots** of how **Apply Lastly Edited Property** (your request) will fit in the workflow:


link to a video:
https://user-images.githubusercontent.com/77115208/204730766-717f55a7-b79a-47b9-84cf-4437e496a414.mp4
![image](https://user-images.githubusercontent.com/77115208/204730947-5bd66e23-efa1-46d9-b382-78483411d83c.png)

p.s. works for object props and map props, giving you great control over how they can be applied to other maps or objects
hope u like it 😊

**Comment by KirilStrezikozin** [comment]
waiting on your comment to assign a ✅ label to this issue)

**Comment by Shlok-Bhakta** [comment]
Thank you so much, it looks great! Can't wait to see the 1.5 update fully put together 😃

**Comment by KirilStrezikozin** [comment]
@Shlok-Bhakta I remember I said that the following functionality is impossible to implement:

> a way to multiselect with ctrl (selects specific objects) and shift (selects everything in the range) then click on settings and have them apply to all of those selected objects. (would also be nice to have a visual indicator that something has been selected)

I think I came up with something...

**Comment by KirilStrezikozin** [comment]
https://user-images.githubusercontent.com/77115208/234628298-19084a25-dac5-4568-9d9c-dffd82d763e9.mp4

here is what I'm talking about. Shift and Ctrl for multiselection. This will allow to multiselect items and set some properties with one click.

**Comment by KirilStrezikozin** [comment]
I'll show more about how it works and the progress... For now, I'll be mentioning commits here for myself to keep track of what's connected with Drag feature

**Comment by KirilStrezikozin** [comment]
there's a lot of work yet to be done for its stability and use cases. Base logic is ready. That all's available in [dev-3.0.0](https://github.com/KirilStrezikozin/BakeMaster-Blender-Addon/tree/dev-3.0.0) branch. If you have any comments or thoughts, don't hesitate to share them)

**Comment by KirilStrezikozin** [comment]
I will close this. Activity will be moved to another thread.
